### PR TITLE
Feat/404

### PR DESCRIPTION
**Pull request by vbvnvmfjg**
* [Trello](https://trello.com/c/kEUUUIc2/42-5-custom-404-page)
* [Preview on page](https://deploy-preview-97--leanwebkit.netlify.com/)

### CODE REVIEW

File: src/client/layouts/error.vue [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<template>
+  <div>
```
Comment (jbmoelker) [inline src/client/layouts/error.vue]: `<div>` not needed

File: src/client/layouts/error.vue [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    <page-layout v-if="error.statusCode === 404" :page="notfound" />
+  </div>
+</template>
+
+<script>
+  import notfoundI18n from '../static/data/404.json'
```
Comment (jbmoelker) [inline src/client/layouts/error.vue]: `notfoundI18n` -> `pageI18n`

File: src/client/layouts/error.vue [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  import { PageLayout } from '../components/'
+
+  export default {
+    components: { PageLayout },
+    props: ['error'],
+    layout: 'default',
+    computed: {
+      locale() { return this.$i18n.locale },
+      notfound() { return notfoundI18n[this.locale] },
```
Comment (jbmoelker) [inline src/client/layouts/error.vue]: `notfound` -> `page`

File: dato.config.js [hunk 4]
```diff
@@ -45,9 +46,10 @@ function appSettingsToJson(app) {
  */
 function redirectsToText (redirects) {
   const redirectToDefaultLocale = `/ /${defaultLocale}/ 301`
+  const redirectToLocalized404s = locales.map(locale => `/${locale}/* /${locale}/404/ 404`).join('\n')
```
Comment (jbmoelker) [inline dato.config.js]:
```js
const redirectToDefaultLocale = `/ /${defaultLocale}/ 301`
const redirectRulesFromCms = redirects
    .map(redirect => `${redirect.from} ${redirect.to} ${redirect.statusCode}`)
const redirectRules404s = locales.map(locale => `/${locale}/* /${locale}/404/ 404`)

return [redirectToDefaultLocale, ...redirectRulesFromCms, ...redirectRules404s].join("\n")
```

File: src/client/layouts/error.vue [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  <div>
+    <page-layout v-if="error.statusCode === 404" :page="notfound" />
+  </div>
+</template>
+
+<script>
+  import notfoundI18n from '../static/data/404.json'
+  import { PageLayout } from '../components/'
+
+  export default {
```
Comment (jbmoelker) [inline src/client/layouts/error.vue]: set `title` in `meta` to `page.title`: https://github.com/nuxt/nuxt.js/blob/master/lib/app/components/nuxt-error.vue#L27

**Review by jbmoelker — CHANGES_REQUESTED** [review]

**Review by Jamerrone — APPROVED** [review]

**Review by jbmoelker — APPROVED** [review]

**Pull request merged**